deployment/datastore/memory_datastore.go

```diff
@@ -82,36 +95,64 @@ func (s *MemoryDataStore[CM]) Merge(other DataStore[CM]) error {
 		}
 	}
 
-	metadataRecords, err := other.ContractMetadata().Fetch()
+	contractMetadataRecords, err := other.ContractMetadata().Fetch()
 	if err != nil {
 		return err
 	}
 
-	for _, record := range metadataRecords {
+	for _, record := range contractMetadataRecords {
 		if err := s.ContractMetadataStore.AddOrUpdate(record); err != nil {
 			return err
 		}
 	}
 
+	// If the env metadata was not set in `other` data store, we don't need to
+	// update it.
+	envMetadata, err := other.EnvMetadata().Get()
+	if err != nil {
+		// Get() will return ErrEnvMetadataNotFound if no record is set. So
+		// we can skip the update in this case.
+		return nil
+	}
```

Comment:
we should specifcally check for the not found error here , in case other error occurs and we do not want to hide it.
```suggestion  
	envMetadata, err := other.EnvMetadata().Get()
	if err != nil {
	        if errors.Is(ErrEnvMetadataNotFound) {
	            return nil
	        }
	        return err
	}
```